domain/agentprovisioner/service/service.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package service
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+
+	"github.com/juju/errors"
+
+	"github.com/juju/juju/core/containermanager"
+	"github.com/juju/juju/core/instance"
+	"github.com/juju/juju/core/model"
+	"github.com/juju/juju/core/providertracker"
+	"github.com/juju/juju/environs"
+	"github.com/juju/juju/environs/config"
+	"github.com/juju/juju/environs/envcontext"
+)
+
+type Provider interface {
+	environs.Networking
+}
+
+type Service struct {
+	modelID        model.UUID
+	providerGetter func(context.Context) (Provider, error)
+	st             State
+}
+
+type State interface {
+	// GetModelConfigKeyValues
+	GetModelConfigKeyValues(context.Context, []string) (map[string]string, error)
+}
+
+func NewService(
+	modelID model.UUID,
+	st State,
+	providerGetter providertracker.ProviderGetter[Provider],
+) *Service {
+	return &Service{
+		providerGetter: providerGetter,
+		st:             st,
+	}
+}
+
+func (s *Service) ContainerManagerConfigForType(
+	ctx context.Context,
+	containerType instance.ContainerType,
+) (containermanager.Config, error) {
+	rval := containermanager.Config{}
+
+	cfg, err := s.st.GetModelConfigKeyValues(ctx, []string{
```

Comment:
We'd also want to get the model config value for "container-networking-method" right?

Reply:
Ah yeah we need to handle that.